### DISCUSSION TITLE
How to upgrade from OS 5.1 to OS 6?

### DISCUSSION THREAD

**Discussion by AnimishY** [elementary/triage#122] [Q&A] [answered]
I have a device of configuration: i3, 4GB RAM, 1TB HDD, Dual Booted to Windows 10 and Elementary 5.1.7 with 125 GB for Elementary OS.

The sole reason to have Dual Boot on such weak and old device was that it was becoming difficult for the device to handle Windows 10 with a web-browser and a video conferencing app together and work without hanging for hours. So to ease my day to day work, I switched to Elementary OS, with no knowledge of Linux at all. I just knew how to make an ISO file to boot able media and install it. 

So I have been using it for months, and now I wish to update to Elementary OS 6, so my basic question is **_how to update to V6 from V5.1.7 without opting for a fresh/clean start, losing all apps and data._** I checked the official website it says nothing for those who already are on the OS, and in the System Settings app, it shows no components is available to update.

Please note I am a 15 year old student with least knowledge of Linux, so please explain all steps, all commands you may refer to. I know it may be a lot extra effort to explain in such manner, and it would really nice of all of you guys to help me out.

Thanks & Regards,
Animish Yadav

**Comment by TomiOhl** [comment, marked as answer]
Hey,
There's no official way to do so now. I beleive they told us 1-2 years ago that 5.x supports upgrading to newer versions, but that doesn't seem to be the case.
So your only good option is to backup your important data from you home folder, install eos from the iso, then restore your data.

**Beware: the following is not approved by elementary team and can go wrong in many ways. Take a backup of your data and follow only on your own responsibility.**

However, it is based on Ubuntu, which supports upgrading between releases. Because of that, there were similar guides for upgrading from elementary 0.4.x to 5.x:
https://www.addictivetips.com/ubuntu-linux-tips/upgrade-to-elementary-os-juno/
https://elementaryos.stackexchange.com/a/16483
The same can be done if you want to update from 5.1.7 to 6.0, too. There are only two differences:

- this time you wanna replace _bionic_ with _focal_ (instead of xenial with bionic). That changes the sed commands.
- _appcenter.list_ doesn't exist for the new version afaik so that can be deleted.

Following both tutorials at the same time (plus doing those two things differently) I could successfully update from Hera to Odin. It worked, but still needed some hours of configuration and fixing things (e.g. had to install pantheon tweaks to set the default theme, had to look for different ppa's than I had before, remove apps that interfered with the new system, etc). Additionally, some default apps (calculator, camera, screenshots) weren't installed and aren't in the app center, but that's not that big of an issue.

**Reply by AnimishY** [reply]
So can we like mark this as a issue, so that the team could work on it if they wish to add this feature to boost their software's functionality by a great extent. 

So maybe bringing it officially to their focus, may make them work for this feature.

And as I said, I am just a student, so maybe I am going wrong on how Open Sourcing works.

And I guess, it is safe for me to just copy my documents and all and take them over to cloud storage [I guess I will need to use every single byte of every cloud storage I have to backup everything, HAHA!] and remove Linux completely off [As I don't know it we can use the installer of new OS to remove another OS] and set it up all over again.

**Reply by TomiOhl** [reply]
I think they are very much aware of this, but feel free to create an issue if you want, nothing bad could happen.
As I remember the Linux Mint team made a small program that did the same steps that I described and it was the official way to update between their LTS (long term support) releases.

Yes, it's perfectly safe to do what you described. However, you don't even have to remove anything before installing, in the installer you will have the option to select the partition where you'd like to install the new version, it will format it automatically.

**Comment by jeremypw** [comment]
I would note that all your important data should be backed up *anyway* e.g. on an external harddisk, whether or not you are upgrading.  You can avoid the need to delete and restore the data by keeping it on a separate partition on your internal harddisk and setting that partition to mount on booting by editing fstab after upgrading.  Obviously, you must use the custom installation option to avoid deleting that partition and it should still be backed up before doing the upgrade. Only use this method if you are comfortable using `gparted` to partition your disk.  You are far less likely to encounter weird bugs if you do a clean install of the new version of elementaryos and then we know that any bugs reported are really due to the os and not the unsupported upgrade process.

Keeping your data in a separate partition also means, if it formatted in a way that both Linux and Windows can read/write, you can share the data between the two oses.

**Reply by AnimishY** [reply]
Thank you, but as I got to know, there is no official option to do it. I actually just backed up my stuff and did a fresh install. For now, I will add apps whenever I need them.

Thank

**Comment by danirabbit** [comment]
Closing as a duplicate of https://github.com/orgs/elementary/discussions/120
